wasmtime: `cargo test` fails to build when cross compiling

👋 Hey,

It looks like we can no longer run `cargo test` in the wasmtime directory when targeting `s390x`,`riscv64` or `aarch64`.

### Test Case

No test case.

### Steps to Reproduce

* `cargo test --target=riscv64gc-unknown-linux-gnu` or  `cargo test --target=s390x-unknown-linux-gnu` or  `cargo test --target=aarch64-unknown-linux-gnu` 

### Expected Results

The tests to build

### Actual Results

```
error[E0432]: unresolved import `cranelift_codegen::isa::x64`
  --> winch/codegen/src/isa/x64/mod.rs:19:30
   |
19 | use cranelift_codegen::{isa::x64::settings as x64_settings, Final, MachBufferFinalized};
   |                              ^^^ could not find `x64` in `isa`

error[E0433]: failed to resolve: could not find `x64` in `isa`
  --> winch/codegen/src/isa/x64/asm.rs:11:10
   |
11 |     isa::x64::{
   |          ^^^ could not find `x64` in `isa`

error[E0432]: unresolved import `cranelift_codegen::isa::x64`
  --> winch/codegen/src/isa/x64/asm.rs:11:10
   |
11 |     isa::x64::{
   |          ^^^ could not find `x64` in `isa`

error[E0432]: unresolved import `cranelift_codegen::isa::x64`
 --> winch/codegen/src/isa/x64/masm.rs:9:30
  |
9 | use cranelift_codegen::{isa::x64::settings as x64_settings, settings, Final, MachBufferFinalized};
  |                              ^^^ could not find `x64` in `isa`

error[E0433]: failed to resolve: could not find `x64` in `isa`
   --> winch/codegen/src/isa/x64/mod.rs:115:67
    |
115 |         Box::new(MachTextSectionBuilder::<cranelift_codegen::isa::x64::Inst>::new(num_funcs))
    |                                                                   ^^^ could not find `x64` in `isa`

warning: unused import: `MachInstEmitState`
  --> winch/codegen/src/isa/x64/asm.rs:18:69
   |
18 |     settings, Final, MachBuffer, MachBufferFinalized, MachInstEmit, MachInstEmitState, Writable,
   |                                                                     ^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

warning: unused import: `MachInstEmit`
  --> winch/codegen/src/isa/x64/asm.rs:18:55
   |
18 |     settings, Final, MachBuffer, MachBufferFinalized, MachInstEmit, MachInstEmitState, Writable,
   |                                                       ^^^^^^^^^^^^

Some errors have detailed explanations: E0432, E0433.
For more information about an error, try `rustc --explain E0432`.
warning: `winch-codegen` (lib) generated 2 warnings
error: could not compile `winch-codegen` (lib) due to 5 previous errors; 2 warnings emitted
```

### Versions and Environment

Wasmtime version or commit: main (ef7af28ef095277f5979dcbc6ff03964c1e0ba12)

Operating system: Linux

Architecture: x86 host / `riscv64gc`,`s390x` or `aarch64` target

### Extra Info

This seems related to winch (cc @saulecabrera)

--- a/winch/codegen/build.rs
+++ b/winch/codegen/build.rs
@@ -3,9 +3,11 @@ fn main() {
         return;
     }
 
-    if cfg!(target_arch = "x86_64") {
+    let arch = std::env::var("CARGO_CFG_TARGET_ARCH").unwrap();
+
+    if arch == "x86_64" {
         println!("cargo:rustc-cfg=feature=\"x64\"");
-    } else if cfg!(target_arch = "aarch64") {
+    } else if arch == "aarch64" {
         println!("cargo:rustc-cfg=feature=\"arm64\"");
     }
     println!("cargo:rerun-if-changed=build.rs");

--- a/winch/codegen/build.rs
+++ b/winch/codegen/build.rs
@@ -9,6 +9,8 @@ fn main() {
         println!("cargo:rustc-cfg=feature=\"x64\"");
     } else if arch == "aarch64" {
         println!("cargo:rustc-cfg=feature=\"arm64\"");
+    } else {
+        println!("cargo:rustc-cfg=feature=\"{arch}\"");
     }
     println!("cargo:rerun-if-changed=build.rs");
 }

